src/clientAuth/clientAuth.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview Adds authentication information to a request and validates
+ * client info.
+ */
+
+const OICCli = require('../init.js').OICCli;
+const utf8 = require('utf8');
+const base64url = require('base64url');
+
+/**
+ * ClientAuthnMethod
+ * Basic Client Authentication Method class.
+ * @class 
+ * @constructor
+ */
+class ClientAuthnMethod {
+  constructor() {}
+
+  construct(params) {
+    throw new Error('Unsupported Operation Exception');
+  }
+}
+
+function validServiceContext(serviceContext, when) {
+  let eta = serviceContext['client_secret_expires_at'] || 0;
```

Comment:
Would it make sense to have `client_secret_expires_at` defined as a member in `ServiceContext`, and access it like `serviceContext.client_secret_expires_at? Or is there a reason to access it like it is now?